x/interchainquery/keeper/msg_server.go

```diff
@@ -57,3 +71,400 @@ func (k msgServer) SubmitQueryResponse(goCtx context.Context, msg *types.MsgSubm
+	bz := k.cdc.MustMarshal(&delegationQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	// query_type := "cosmos.bank.v1beta1.Query/AllBalances"
+	// balanceQuery := banktypes.QueryAllBalancesRequest{Address: msg.Address}
+	// bz, err := k.cdc.Marshal(&balanceQuery)
+
+	k.Keeper.MakeRequest(
+		ctx,
+		ConnectionId,
+		ChainId,
+		query_type,
+		bz,
+		sdk.NewInt(25),
+		types.ModuleName,
+		cb,
+	)
+
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			sdk.EventTypeMessage,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.AttributeValueCategory),
+			sdk.NewAttribute(sdk.AttributeKeyAction, types.AttributeValueQuery),
+			sdk.NewAttribute(types.AttributeKeyQueryId, GenerateQueryHash(ConnectionId, ChainId, query_type, bz)),
+			sdk.NewAttribute(types.AttributeKeyChainId, ChainId),
+			sdk.NewAttribute(types.AttributeKeyConnectionId, ConnectionId),
+			sdk.NewAttribute(types.AttributeKeyType, query_type),
+			sdk.NewAttribute(types.AttributeKeyHeight, "0"),
+			sdk.NewAttribute(types.AttributeKeyRequest, hex.EncodeToString(bz)),
+		),
+	})
+
+	// return; usually a response object or nil
+	return &types.MsgQueryExchangerateResponse{}, nil
+}
+
+func (k msgServer) QueryDelegatedbalance(goCtx context.Context, msg *types.MsgQueryDelegatedbalance) (*types.MsgQueryDelegatedbalanceResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	ChainId := msg.ChainId
+	// TODO Check ChainId is supported by Stride, try using this approach https://github.com/ingenuity-build/quicksilver/blob/ea71f23c6ef09a57e601f4e544c4be9693f5ba81/x/interchainstaking/keeper/msg_server.go#L37
+
+	// Parse Address addr
+	// TODO should this be Address, not Caller? changed temporarily to suppress error
+	_, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		panic(err)
+	}
+
+	// get zone
+	hz, found := k.StakeibcKeeper.GetHostZone(ctx, ChainId)
+	if found {
+		fmt.Errorf("invalid chain id, zone for \"%s\" already registered", ChainId)
+	}
+	ConnectionId := hz.ConnectionId
+
+	var cb Callback = func(k Keeper, ctx sdk.Context, args []byte, query types.Query) error {
+		queryRes := banktypes.QueryAllBalancesResponse{}
+		err := k.cdc.Unmarshal(args, &queryRes)
+		if err != nil {
+			k.Logger(ctx).Error("Unable to unmarshal balances info for zone", "err", err)
+			return err
+		}
+		// TODO get denom dynamically
+		balance := int32(queryRes.Balances.AmountOf("uatom").Int64())
+
+		// Set delegation account balance to ICQ result
+		hz, found := k.StakeibcKeeper.GetHostZone(ctx, ChainId)
+		if found {
+			fmt.Errorf("invalid chain id, zone for \"%s\" already registered", ChainId)
+		}
+
+		da := hz.DelegationAccount
+		delegationAccount := stakeibctypes.ICAAccount{Address: da.Address,
+			Balance:          balance, // <== updated
+			DelegatedBalance: da.DelegatedBalance,
+			Delegations:      da.Delegations,
+			Target:           da.Target,
+		}
+
+		// set the zone
+		zone := stakeibctypes.HostZone{
```

Comment:
Same comment as above for the `zone`, can we just update the field on `hz`? 

Sorry, I think I may have sent you this code snippet, which was misleading